crates/vm/db.rs

```diff
@@ -1,8 +1,71 @@
-use ethrex_common::types::BlockHash;
+use crate::EvmError;
+use bytes::Bytes;
+use dyn_clone::DynClone;
+use ethrex_common::{
+    types::{AccountInfo, BlockHash, ChainConfig},
+    Address, H256, U256,
+};
 use ethrex_storage::Store;
 
+pub trait VmDatabase: Send + Sync + DynClone {
+    fn get_account_info(&self, address: Address) -> Result<Option<AccountInfo>, EvmError>;
+    fn get_storage_slot(&self, address: Address, key: H256) -> Result<Option<U256>, EvmError>;
+    fn get_block_hash(&self, block_number: u64) -> Result<Option<H256>, EvmError>;
+    fn get_chain_config(&self) -> ChainConfig;
+    fn get_account_code(&self, code_hash: H256) -> Result<Option<Bytes>, EvmError>;
+}
+
+dyn_clone::clone_trait_object!(VmDatabase);
+
+pub struct DynVmDatabase(pub Box<dyn VmDatabase + Send + Sync + 'static>);
```

Comment:
Could this be a type alias instead of a struct?

Reply:
I think it is better as it makes it more of a helper that can be used to avoid using the full type definition but that can be ignored on cases where the compiler can infer the type correctly (or just not used if the user prefers specifying the types explicitly). Rather than a wrapper structure that must be used